src/components/DataDrawer/index.tsx

```diff
@@ -9,18 +9,27 @@ import {
 } from '@material-ui/core';
 import DataTable from './DataTable';
 import { getIsShowing, hideTable } from '../../context/tableStateSlice';
+import {
+  isDataTableDrawerActiveSelector,
+  hideDataTableDrawer,
+} from '../../context/analysisResultStateSlice';
 
 function DataDrawer({ classes }: DataDrawerProps) {
   const tableIsShowing = useSelector(getIsShowing);
+  const analysisTableIsShowing = useSelector(isDataTableDrawerActiveSelector);
+  const isShowing = tableIsShowing || analysisTableIsShowing;
+
   const dispatch = useDispatch();
   const handleClose = () => {
     dispatch(hideTable());
+    dispatch(hideDataTableDrawer());
   };
 
   return (
-    <Drawer anchor="left" open={tableIsShowing} onClose={handleClose}>
+    <Drawer anchor="left" open={isShowing} onClose={handleClose}>
       <div className={classes.drawerContent}>
-        <DataTable maxResults={1000} />
+        {tableIsShowing ||
+          (analysisTableIsShowing && <DataTable maxResults={1000} />)}
```

Comment:
This is somewhat confusing. Think we can combine this to one conditional expression?

Reply:
Did just that. Check the new implementation below

```javascript
const isShowing = tableIsShowing || analysisTableIsShowing;
...
{isShowing && <DataTable maxResults={1000} />}
```